src/OrchardCore.Modules/OrchardCore.PublishLater/Migrations.cs

```diff
@@ -1,59 +1,84 @@
 using System;
 using OrchardCore.ContentManagement.Metadata;
 using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.ContentManagement.Records;
 using OrchardCore.Data.Migration;
 using OrchardCore.PublishLater.Indexes;
 using OrchardCore.PublishLater.Models;
 using YesSql.Sql;
 
-namespace OrchardCore.PublishLater
+namespace OrchardCore.PublishLater;
+
+public class Migrations : DataMigration
 {
-    public class Migrations : DataMigration
+    private readonly IContentDefinitionManager _contentDefinitionManager;
+
+    public Migrations(IContentDefinitionManager contentDefinitionManager)
     {
-        private readonly IContentDefinitionManager _contentDefinitionManager;
+        _contentDefinitionManager = contentDefinitionManager;
+    }
 
-        public Migrations(IContentDefinitionManager contentDefinitionManager)
-        {
-            _contentDefinitionManager = contentDefinitionManager;
-        }
+    public int Create()
+    {
+        _contentDefinitionManager.AlterPartDefinition(nameof(PublishLaterPart), builder => builder
+            .Attachable()
+            .WithDescription("Adds the ability to schedule content items to be published at a given future date and time."));
 
-        public int Create()
-        {
-            _contentDefinitionManager.AlterPartDefinition(nameof(PublishLaterPart), builder => builder
-                .Attachable()
-                .WithDescription("Adds the ability to schedule content items to be published at a given future date and time."));
-
-            SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
-                .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
-
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
-                    "DocumentId",
-                    nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
-
-            // Shortcut other migration steps on new content definition schemas.
-            return 2;
-        }
-
-        // This code can be removed in a later version.
-        public int UpdateFrom1()
-        {
-            // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
-            // this because dropping an index and altering a column don't work on all providers.
+        SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
+            .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
+                "DocumentId",
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
```

Comment:
Idem and with the new included column `Id`.

Reply:
Why?

Reply:
As we did for ArchiveLater and that I commented   https://github.com/OrchardCMS/OrchardCore/pull/10935#issuecomment-1221855101

See also https://github.com/OrchardCMS/OrchardCore/pull/10935#issuecomment-1236189580 for more details about non clustered indexes.

So here because we want to use `QueryIndex()` that returns all columns of the index Table including the `Id` column, good to also include the `Id` column in this non clustered index (which is an inner index of the index Table), so that we have a better chance to not hit the DB clusters.